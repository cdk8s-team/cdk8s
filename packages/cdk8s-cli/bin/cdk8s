#!/usr/bin/env node
require('./cdk8s.js');