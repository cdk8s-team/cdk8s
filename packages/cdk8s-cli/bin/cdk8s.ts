import * as yargs from 'yargs';

yargs
  .commandDir('cmds')
  .demandCommand()
  .help()
  .argv;
