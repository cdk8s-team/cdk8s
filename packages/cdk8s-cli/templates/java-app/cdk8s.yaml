language: java
app:  mvn exec:java -Dexec.mainClass="com.mycompany.app.Main"
imports:
  - k8s