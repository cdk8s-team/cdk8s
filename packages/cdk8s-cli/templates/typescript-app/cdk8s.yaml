language: typescript
app: node main.js
imports:
  - k8s
