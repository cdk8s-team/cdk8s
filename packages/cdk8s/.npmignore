# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
/.eslintrc.json
/.github
/.idea
/.projen
/.projenrc.js
/.versionrc.json
/.vscode
/coverage
/src
/test
/test-reports/
/tsconfig.eslint.json
/tsconfig.jest.json
/tsconfig.json
dist
junit.xml
!.jsii
!/lib
!/lib/**/*.d.ts
!/lib/**/*.js
