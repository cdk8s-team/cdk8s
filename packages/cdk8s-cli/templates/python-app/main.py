#!/usr/bin/env python
from constructs import Construct
from cdk8s import App, Chart


class MyChart(Chart):
    def __init__(self, scope: Construct, id: str):
        super().__init__(scope, id)

        # define resources here


app = App()
MyChart(app, "{{ $base }}")

app.synth()
