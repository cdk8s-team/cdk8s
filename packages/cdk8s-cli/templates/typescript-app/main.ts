import { Construct } from '@aws-cdk/core';
import { App, Chart } from 'cdk8s';

class MyChart extends Chart {
  constructor(scope: Construct, name: string) {
    super(scope, name);

    // define constructs here
  }
}

const app = new App();
new MyChart(app, '{{ $base }}');
app.synth();
