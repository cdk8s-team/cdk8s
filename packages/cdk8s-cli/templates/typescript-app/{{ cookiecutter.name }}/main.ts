import { App, Construct } from '@aws-cdk/core';
import { Chart } from '@awslabs/cdk8s';

class MyChart extends Chart {
  constructor(scope: Construct, name: string) {
    super(scope, name);

    // define constructs here
  }
}

const app = new App({ outdir: 'dist' });
new MyChart(app, '{{ cookiecutter.name }}');
app.synth();
