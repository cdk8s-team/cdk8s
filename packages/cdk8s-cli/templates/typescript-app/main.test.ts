import {MyChart} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new MyChart(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
