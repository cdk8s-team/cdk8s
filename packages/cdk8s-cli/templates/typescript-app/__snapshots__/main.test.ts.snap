// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `Array []`;
